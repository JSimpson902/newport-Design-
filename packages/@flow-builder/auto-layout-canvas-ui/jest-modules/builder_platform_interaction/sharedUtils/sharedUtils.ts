const sharedUtils = jest.requireActual('builder_platform_interaction_mocks/sharedUtils/sharedUtils');
const commands = jest.requireActual('builder_platform_interaction/sharedUtils/commands');
const lwcUtils = jest.requireActual('builder_platform_interaction/sharedUtils/lwcUtils');

const mockKeyboardInteractionUtils = jest.requireActual(
    'builder_platform_interaction_mocks/sharedUtils/keyboardInteractionUtils'
);
const auraUtils = jest.requireActual('builder_platform_interaction/sharedUtils/auraUtils');
const customIconUtils = jest.requireActual('builder_platform_interaction/sharedUtils/customIconUtils');

let keyboardInteractionUtils = jest.requireActual('builder_platform_interaction/sharedUtils/keyboardInteractionUtils');
const { KeyboardInteractions, withKeyboardInteractions } = mockKeyboardInteractionUtils;

keyboardInteractionUtils = { ...keyboardInteractionUtils, KeyboardInteractions, withKeyboardInteractions };

const { loggingUtils, storeUtils, commonUtils } = sharedUtils;

const Keys = keyboardInteractionUtils.Keys;

const focusUtils = {
    getElementWithFocus: jest.fn(),
    restoreFocus: jest.fn(),
    clearStashFocus: jest.fn(),
    stashFocus: jest.fn()
};

const appGuidanceUtils = jest.requireActual('builder_platform_interaction/sharedUtils/appGuidanceUtils');

export {
    loggingUtils,
    keyboardInteractionUtils,
    commands,
    storeUtils,
    commonUtils,
    lwcUtils,
    auraUtils,
    Keys,
    focusUtils,
    customIconUtils,
    appGuidanceUtils
};
