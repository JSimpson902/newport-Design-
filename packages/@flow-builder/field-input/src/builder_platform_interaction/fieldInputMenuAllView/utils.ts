import { getFlowElementsApi, getRulesApi } from 'builder_platform_interaction/fieldInputMenuDataApi';
import { filterAndMapToMenuItems } from 'builder_platform_interaction/fieldInputUtils';
import { ELEMENT_TYPE } from 'builder_platform_interaction/flowMetadata';
import { getLHSTypes, RULE_TYPES } from 'builder_platform_interaction/ruleLib';

/**
 * Get the sections for the "All" view
 *
 * @returns The "All" view sections
 */
export function getViewAllSections(): FieldInput.MenuSection[] {
    const config: MenuConfig = {
        sortField: 'label',
        activePicklistValues: [],
        traversalConfig: { isEnabled: true },
        filter: {
            includeNewResource: true,
            allowGlobalConstants: false,
            showSystemVariables: true,
            showGlobalVariables: true,
            shouldBeWritable: false
        }
    };

    // TODO: remove hardcoded
    const ruleType = RULE_TYPES.ASSIGNMENT;
    const elementType = ELEMENT_TYPE.DECISION;

    const rules = getRulesApi(ruleType, elementType);

    const flowELements = getFlowElementsApi();

    // TODO: find better way to wait for rules to load and remove this
    if (rules != null) {
        return filterAndMapToMenuItems(flowELements, getLHSTypes(ELEMENT_TYPE.DECISION, rules), config);
    }

    return [];
}
