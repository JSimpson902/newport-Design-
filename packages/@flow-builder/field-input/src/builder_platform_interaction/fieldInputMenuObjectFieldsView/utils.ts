/**
 *
 * @param apiData TBD
 * @returns TBD
 */
export function getObjectFieldsViewSections(apiData: GetObjectInfoApiData): FieldInput.MenuSection[] {
    const menuItems: FieldInput.MenuItem[] = Object.values(apiData.fields).map((field) => ({
        ...field,
        type: field.dataType,
        name: field.apiName,
        value: field.apiName
    }));

    return [{ items: menuItems, key: 'section-default', label: undefined }];
}
