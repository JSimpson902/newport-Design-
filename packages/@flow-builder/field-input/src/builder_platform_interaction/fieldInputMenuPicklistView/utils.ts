/**
 *
 * @param apiData TBD
 * @returns TBD
 */
export function getPicklistViewSection(apiData: GetPicklistValuesApiData): FieldInput.MenuSection {
    return {
        key: 'section-default',
        items: (apiData.values || []).map((value) => ({
            ...value,
            label: value.value,
            name: value.value,
            key: value.value
        }))
    };
}
