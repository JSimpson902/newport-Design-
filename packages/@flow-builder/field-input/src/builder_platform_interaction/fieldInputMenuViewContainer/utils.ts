const viewDispatchMap = {
    All: { isAllView: true },
    ObjectFields: { isObjectFieldsView: true },
    PicklistValues: { isPicklistValuesView: true }
} as const;

/**
 * Augments a MenuItemView with a `isXX` dispatch property
 *
 * @param view - The menu item view
 * @returns The augmented menu item view
 */
export const getViewProps = (view: FieldInput.MenuItemView) => {
    const viewDispatch = viewDispatchMap[view.type];

    return viewDispatch != null ? { ...view, ...viewDispatch } : { missingViewType: view.type };
};
