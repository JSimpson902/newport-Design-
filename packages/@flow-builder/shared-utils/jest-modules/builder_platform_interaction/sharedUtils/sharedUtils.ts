import * as loggingUtils from './loggingUtils/loggingUtils';
import * as keyboardInteractionUtils from './keyboardInteractionUtils/keyboardInteractionUtils';

import * as commands from './commands/commands';
import * as storeUtils from './storeUtils/storeUtils';
import * as commonUtils from './commonUtils/commonUtils';
export { loggingUtils, keyboardInteractionUtils, commands, storeUtils, commonUtils };
