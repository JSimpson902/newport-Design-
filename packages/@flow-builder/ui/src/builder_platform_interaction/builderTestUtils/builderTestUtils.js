export * from './screenEditorTestUtils.js';
export * from './commonTestUtils';
export * from './deepFindMatchers';
export * from './useAdvancedOptionsTestUtils';
export * from './goldObjectMatchers';
export * from './domTestUtils';
export * from './complexTypeTestUtils';
