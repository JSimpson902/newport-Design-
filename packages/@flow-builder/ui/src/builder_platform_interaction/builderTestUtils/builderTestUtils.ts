export * from './combobox';
export * from './commonTestUtils';
export * from './complexTypeTestUtils';
export * from './deepFindMatchers';
export * from './domTestUtils';
export * from './events';
export * from './goldObjectMatchers';
export * from './manuallyAssignVariablesCheckboxTestUtils';
export * from './screenEditorTestUtils';
export * from './selectors';

// @sa11y testing appears to take some extra time (beyond the given 5000ms default jest timeout)
export const ACCESSIBILITY_TEST_TIMEOUT_IN_MS = 15000;

/**
 * Mocks a label in jest
 *
 * @param jest - The jest instance
 * @param labelName - The label name
 * @returns the mock label
 */
export function mockLabel(jest, labelName: string) {
    jest.mock(
        `@salesforce/label/${labelName}`,
        () => {
            return { default: `${labelName} mock value` };
        },
        { virtual: true }
    );

    return `${labelName} mock value`;
}
