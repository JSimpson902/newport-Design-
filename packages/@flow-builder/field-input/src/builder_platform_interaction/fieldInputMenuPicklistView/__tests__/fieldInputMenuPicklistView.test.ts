import { createComponent } from 'builder_platform_interaction/builderTestUtils';
import { lwcUtils } from 'builder_platform_interaction/sharedUtils';
import { getPicklistValuesViewSection } from '../utils';

const tag = 'builder_platform_interaction-field-input-menu-picklist-view';

const apiData: GetPicklistValuesApiData = {
    values: [
        {
            label: 'Advertisement',
            value: 'Advertisement'
        },
        {
            label: 'Employee Referral',
            value: 'Employee Referral'
        }
    ]
};

const mockApiResponse = {
    data: apiData
};

jest.mock('builder_platform_interaction/fieldInputMenuSectionItem', () =>
    jest.requireActual('builder_platform_interaction_mocks/fieldInputMenuSectionItem')
);

jest.mock('builder_platform_interaction/fieldInputMenuDataApi', () => {
    return {
        getPicklistValuesApi: () => {
            return Promise.resolve(mockApiResponse);
        }
    };
});

jest.mock('../utils', () => {
    return {
        getPicklistValuesViewSection: jest.fn((apiData) =>
            jest.requireActual('../utils').getPicklistValuesViewSection(apiData)
        )
    };
});

const createComponentUnderTest = async (props) => {
    const overrideProps = {};

    return createComponent(tag, props, overrideProps);
};

const selectors = {
    ul: 'ul'
};

describe('Field Input Menu Picklist View Tests', () => {
    let cmp;
    let dom;

    beforeEach(async () => {
        cmp = await createComponentUnderTest({
            info: {
                recordTypeId: '012',
                fieldApiName: 'Account.Advertisement'
            }
        });
        dom = lwcUtils.createDomProxy(cmp, selectors);
    });

    it('sanity', async () => {
        expect(cmp).toBeTruthy();
        expect(getPicklistValuesViewSection).toHaveBeenCalledTimes(1);
        expect(dom.all.ul.length).toEqual(1);
    });

    it('utils > getPicklistValuesViewSection', async () => {
        expect(getPicklistValuesViewSection(mockApiResponse.data)).toMatchSnapshot();
    });
});
