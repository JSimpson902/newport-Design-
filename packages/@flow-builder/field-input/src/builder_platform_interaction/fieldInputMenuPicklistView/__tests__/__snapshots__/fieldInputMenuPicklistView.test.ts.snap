// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Input Menu Picklist View Tests utils > getPicklistViewSection 1`] = `
Object {
  "items": Array [
    Object {
      "key": "Advertisement",
      "label": "Advertisement",
      "name": "Advertisement",
      "value": "Advertisement",
    },
    Object {
      "key": "Employee Referral",
      "label": "Employee Referral",
      "name": "Employee Referral",
      "value": "Employee Referral",
    },
  ],
  "key": "section-default",
}
`;
